### PR TITLE
Move `forc`'s LSP support into a dedicated `forc-lsp` plugin crate

### PR DESCRIPTION
**Pull request by mitchmindtree**
**NOTE: This PR is Based on #1178 - we should land that first.** I've opened this draft as a working demonstration of plugin support implemented in #1178.

This creates a dedicated `forc-lsp` plugin crate. With `forc`'s new
plugin support, having this plugin installed and available via `PATH`
enables the original behaviour of starting the sway language server via
`forc lsp <args>`.

This also acts as a local test for `forc` plugin support and as a simple
example of how to implement a `forc` plugin.

The old `lsp` and `forc_lsp` modules have been removed from `forc` in
favour of this new `forc-lsp` plugin.

### CODE REVIEW

**Comment by mitchmindtree** [comment]
See [this comment](https://github.com/FuelLabs/sway/pull/1178#issuecomment-1091105940) for a demonstration of how to install and use this forc plugin.

File: forc-lsp/Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,15 @@
+[package]
+name = "forc-lsp"
+version = "0.9.0"
```
Comment (adlerjohn) [inline forc-lsp/Cargo.toml:3]:
```suggestion
version = "0.9.2"
```

**Review by adlerjohn — COMMENTED** [review]
While we're here, could you use major.minor for non-Fuel deps in this new crate instead of major.minor.patch?

File: forc-lsp/Cargo.toml [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+edition = "2021"
+homepage = "https://fuel.network/"
+license = "Apache-2.0"
+repository = "https://github.com/FuelLabs/sway"
+description = "A simple `forc` plugin for starting the sway language server."
+
+[dependencies]
+anyhow = "1"
+clap = { version = "3", features = ["derive"] }
+sway-lsp = { version = "0.9", path = "../sway-lsp" }
```
Comment (adlerjohn) [inline forc-lsp/Cargo.toml:14]: Wait no, Fuel deps still need an exact version otherwise the publishing action won't work :joy:

Comment (mitchmindtree) [reply forc-lsp/Cargo.toml:14]: agh of course!

Comment (adlerjohn) [reply forc-lsp/Cargo.toml:14]: We really need to augment https://github.com/FuelLabs/cargo-toml-lint to check for that!

**Review by JoshuaBatty — APPROVED** [review]
utACK. 

Although users will need to install the LSP plugin manually for the time being, this is probably good motivation to also look into shipping the LSP binary with the VScode plugin itself.

**Pull request merged**